internal/ui/login/handler/external_login_handler.go

```diff
@@ -106,13 +118,75 @@ func (l *Login) handleExternalLoginCallback(w http.ResponseWriter, r *http.Reque
 		l.renderError(w, r, authReq, err)
 		return
 	}
-	provider := l.getRPConfig(w, r, authReq, idpConfig, EndpointExternalLoginCallback)
-	tokens, err := rp.CodeExchange(r.Context(), data.Code, provider)
+	if idpConfig.IDPConfigOIDCView != nil {
+		provider := l.getRPConfig(w, r, authReq, idpConfig, EndpointExternalLoginCallback)
+		tokens, err := rp.CodeExchange(r.Context(), data.Code, provider)
+		if err != nil {
+			l.renderLogin(w, r, authReq, err)
+			return
+		}
+		l.handleExternalUserAuthenticated(w, r, authReq, idpConfig, userAgentID, tokens)
+	}
+	if idpConfig.IDPConfigAuthConnectorView != nil {
+		l.handleExternalLoginCallbackAuthConnector(w, r, authReq, data, idpConfig)
+	}
+}
+
+func (l *Login) handleExternalLoginCallbackAuthConnector(w http.ResponseWriter, r *http.Request, authReq *domain.AuthRequest, data *externalIDPCallbackData, idpConfig *iam_model.IDPConfigView) {
```

Comment:
maybe it would make sense to split this func, because it makes a lot of things and is really long

Reply:
split into two methods